datasets/fr/amf_regulatory_sanctions/crawler.py

```diff
@@ -0,0 +1,138 @@
+import requests
+from typing import Any, Generator, Optional, Tuple, Union
+from zavod import Context
+from datetime import datetime
+import re
+
+# Regex patterns for cleaning theme
+theme_patterns = {
+    re.compile(r"Manquement d&#039;initié"): "Manquement d'initié",
+    re.compile(r"Obligation d&#039;information"): "Obligation d'information",
+    re.compile(r"Manipulation de marché"): "Manipulation de marché",
+    re.compile(r"CIF, CIP ou autres prestataires"): "CIF, CIP ou autres prestataires",
+    re.compile(r"Instrument financier"): "Instrument financier",
+    re.compile(r"Produit d&#039;épargne collective"): "Produit d'épargne collective",
+    re.compile(r"Obligations professionnelles"): "Obligations professionnelles",
+    re.compile(r"Infrastructure de marché"): "Infrastructure de marché",
+    re.compile(r"Procédure"): "Procédure",
+    re.compile(r"PSI"): "PSI",
+}
+
+
+def get_value(
+    data: dict, keys: Tuple[str, ...], default: Optional[Any] = None
+) -> Union[Optional[Any], Any]:
+    for key in keys:
+        val = data.pop(key, None)
+        if val is not None:
+            return val
+    return default
+
+
+def parse_json(context: Context) -> Generator[dict, None, None]:
+    response = requests.get(
+        "https://www.amf-france.org/fr/rest/listing_sanction/91,184,183,325,90,461,89,242,86,462,181/all/all?t=1722860865328&_=1722860865172"
+    )
+    if response.status_code != 200:
+        context.log.error(
+            f"Failed to fetch data: {response.status_code}, {response.text}"
+        )
+        return
```

Comment:
context.fetch_json has this checking built in

```suggestion
    response = context.fetch_json(
        "https://www.amf-france.org/fr/rest/listing_sanction/91,184,183,325,90,461,89,242,86,462,181/all/all?t=1722860865328&_=1722860865172"
    )
    if response.status_code != 200:
        context.log.error(
            f"Failed to fetch data: {response.status_code}, {response.text}"
        )
        return
```